lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -5600,8 +5600,14 @@ using namespace Js;
         Var lowerValue = nullptr, upperValue = nullptr;
         T lowerExists, upperExists;
         const char16* methodName;
-        bool isTypedArrayEntryPoint = typedArrayBase != nullptr;
 
+        // If we came from Array.prototype.map and source object is not a JavascriptArray, source could be a TypedArray
+        if (typedArrayBase == nullptr && pArr == nullptr && VarIs<TypedArrayBase>(obj))
```

Comment:
Is the effect of moving this up just to change the `methodName` below, or is there something else?

Reply:
yes

---
In reply to: [323436676](https://github.com/microsoft/ChakraCore/pull/6280#discussion_r323436676) [](ancestors = 323436676)

Reply:
Would this name still be correct if we did something like `Array.prototype.reverse.call(new Int8Array(2))` ? We're going to think the entry point was `%TypedArray%.prototype.reverse`, right?

Reply:
Yes, it will be %TypedArray%.prototype.reverse. Is that not correct?

---
In reply to: [323489092](https://github.com/microsoft/ChakraCore/pull/6280#discussion_r323489092) [](ancestors = 323489092)

Reply:
If you call `Array.prototype.reverse` (regardless of what you call it with) it should probably report that name in any error messages.

Reply:
Yeah. The error message text is not spec'd but we want the name reported to match the actual entrypoint called - they're differently named functions in the spec, after all.

Reply:
Unless, there is a strong preference on the error message I will make this change. Let me know.

---
In reply to: [323510851](https://github.com/microsoft/ChakraCore/pull/6280#discussion_r323510851) [](ancestors = 323510851)

Reply:
I think we should probably stick with the current method of determining the `methodName`?

Reply:
But that does cause a functional issue. In this particular case we will fail prematurely on Line#5620 as we will not find the method "Array.prototype.reverse"

const var_0 = new Uint8Array([1, 2, 3, 4]);
Object.defineProperty(var_0, 'length', { value: 4398046507008 });
Array.prototype.reverse.call(var_0);


---
In reply to: [324277790](https://github.com/microsoft/ChakraCore/pull/6280#discussion_r324277790) [](ancestors = 324277790)